Userland/Utilities/markdown-check.cpp

```diff
@@ -239,13 +242,55 @@ RecursionDecision MarkdownLinkage::visit(Markdown::Text::LinkNode const& link_no
     return RecursionDecision::Recurse;
 }
 
+static ErrorOr<String> generate_link_graph(HashMap<NonnullRefPtr<Manual::PageNode const>, Vector<NonnullRefPtr<Manual::PageNode const>>> page_links)
+{
+    auto const header = "digraph manpage_links {\n"sv;
+    StringBuilder builder;
+    TRY(builder.try_append(header));
+
+    // Not displayed to the user.
+    HashMap<NonnullRefPtr<Manual::PageNode const>, String> page_identifiers;
+
+    for (auto const& page : page_links.keys()) {
+        auto path = TRY(page->path());
+        StringBuilder identifier_builder;
+        // Only allow alphanumerics, replace everything else with underscores.
+        for (auto const& character : path.code_points()) {
+            if (AK::is_ascii_alphanumeric(character))
+                TRY(identifier_builder.try_append_code_point(character));
+            else
+                TRY(identifier_builder.try_append('_'));
+        }
+        auto const identifier = TRY(identifier_builder.to_string());
+        TRY(builder.try_appendff("{} [label=\"{}({})\"];\n", identifier, TRY(page->name()), page->section_number()));
+        TRY(page_identifiers.try_set(page, identifier));
+    }
+
+    for (auto const& from_page_list : page_links) {
+        auto const& from_page = from_page_list.key;
+        for (auto const& to_page : from_page_list.value) {
+            auto const to_page_identifier = page_identifiers.get(to_page);
+            // Target page doesn't actually exist; it's probably an ignored page.
+            if (!to_page_identifier.has_value())
+                continue;
+            TRY(builder.try_appendff("{} -> {};\n", page_identifiers.get(from_page).release_value(), page_identifiers.get(to_page).release_value()));
```

Comment:
I'd only use `release_value()` if we actually need ownership, this seems fine:

```suggestion
            TRY(builder.try_appendff("{} -> {};\n", *page_identifiers.get(from_page), *page_identifiers.get(to_page)));
```